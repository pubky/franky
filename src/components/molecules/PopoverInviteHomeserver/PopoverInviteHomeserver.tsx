import * as Atoms from '@/atoms';
import * as Libs from '@/libs';

interface PopoverInviteHomeserverProps {
  className?: string;
}

export function PopoverInviteHomeserver({ className = 'hover:bg-brand/10' }: PopoverInviteHomeserverProps) {
  return (
    <Atoms.Popover>
      <Atoms.PopoverTrigger asChild>
        <Atoms.Button variant="ghost" size="icon" className={className}>
          <Libs.CircleHelp className="h-4 w-4 text-white" data-testid="circle-help-icon" />
        </Atoms.Button>
      </Atoms.PopoverTrigger>
      <Atoms.PopoverContent className="w-[327px]">
        <Atoms.Container className="flex-col gap-6 px-3 py-2">
          <Atoms.Container className="flex-col gap-2">
            <Atoms.Heading level={4} size="sm" className="text-popover-foreground">
              Don&apos;t have an invite yet?
            </Atoms.Heading>
            <Atoms.Typography size="sm" className="text-muted-foreground font-medium">
              Ask the Pubky team for your invite code to access the Pubky homeserver.
            </Atoms.Typography>
            <Atoms.Typography size="sm" className="text-muted-foreground font-medium">
              A homeserver is a storage provider that hosts your pubky-app-related data. Prefer to use another provider
              or host yourself? Support for custom homeservers is coming soon.
            </Atoms.Typography>
          </Atoms.Container>
          <Atoms.Container className="flex-row gap-4">
            <Atoms.Link href={Libs.SocialLinks.EMAIL} className="text-muted-foreground hover:text-brand">
              <Libs.Mail className="h-6 w-6" />
            </Atoms.Link>
            <Atoms.Link href={Libs.SocialLinks.TWITTER} className="text-muted-foreground hover:text-brand">
              <Libs.XTwitter className="h-6 w-6" />
            </Atoms.Link>
            <Atoms.Link href={Libs.SocialLinks.TELEGRAM} className="text-muted-foreground hover:text-brand">
              <Libs.Telegram className="h-6 w-6" />
            </Atoms.Link>
          </Atoms.Container>
        </Atoms.Container>
      </Atoms.PopoverContent>
    </Atoms.Popover>
  );
}
