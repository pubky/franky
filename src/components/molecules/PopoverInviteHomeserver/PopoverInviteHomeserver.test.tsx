import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import { PopoverInviteHomeserver } from './PopoverInviteHomeserver';

// Mock libs - use actual utility functions and icons from lucide-react
vi.mock('@/libs', async (importOriginal) => {
  const actual = await importOriginal<typeof import('@/libs')>();
  return { ...actual };
});

// Mock atoms
vi.mock('@/atoms', () => ({
  Popover: ({ children }: { children: React.ReactNode }) => <div data-testid="popover">{children}</div>,
  PopoverTrigger: ({ children, asChild }: { children: React.ReactNode; asChild?: boolean }) => (
    <div data-testid="popover-trigger" data-as-child={asChild}>
      {children}
    </div>
  ),
  PopoverContent: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="popover-content" className={className}>
      {children}
    </div>
  ),
  Button: ({
    children,
    variant,
    size,
    className,
  }: {
    children: React.ReactNode;
    variant?: string;
    size?: string;
    className?: string;
  }) => (
    <button data-testid={`button-${variant || 'default'}`} data-size={size} className={className}>
      {children}
    </button>
  ),
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" className={className}>
      {children}
    </div>
  ),
  Heading: ({
    children,
    level,
    size,
    className,
  }: {
    children: React.ReactNode;
    level?: number;
    size?: string;
    className?: string;
  }) => {
    const Tag = `h${level || 1}` as keyof JSX.IntrinsicElements;
    return (
      <Tag data-testid={`heading-${level || 1}`} data-size={size} className={className}>
        {children}
      </Tag>
    );
  },
  Typography: ({ children, size, className }: { children: React.ReactNode; size?: string; className?: string }) => (
    <p data-testid="typography" data-size={size} className={className}>
      {children}
    </p>
  ),
  Link: ({ children, href, className }: { children: React.ReactNode; href: string; className?: string }) => (
    <a data-testid="link" href={href} className={className}>
      {children}
    </a>
  ),
}));

describe('PopoverInviteHomeserver - Snapshots', () => {
  it('matches snapshot for default PopoverInviteHomeserver', () => {
    const { container } = render(<PopoverInviteHomeserver />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for PopoverInviteHomeserver with custom className', () => {
    const { container } = render(<PopoverInviteHomeserver className="custom-homeserver-style" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
