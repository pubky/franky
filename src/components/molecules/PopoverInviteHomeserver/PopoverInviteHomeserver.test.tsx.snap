// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopoverInviteHomeserver - Snapshots > matches snapshot for PopoverInviteHomeserver with custom className 1`] = `
<div
  data-testid="popover"
>
  <div
    data-as-child="true"
    data-testid="popover-trigger"
  >
    <button
      class="custom-homeserver-style"
      data-size="icon"
      data-testid="button-ghost"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-circle-help h-4 w-4 text-white"
        data-testid="circle-help-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <path
          d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
        />
        <path
          d="M12 17h.01"
        />
      </svg>
    </button>
  </div>
  <div
    class="w-[327px]"
    data-testid="popover-content"
  >
    <div
      class="flex-col gap-6 px-3 py-2"
      data-testid="container"
    >
      <div
        class="flex-col gap-2"
        data-testid="container"
      >
        <h4
          class="text-popover-foreground"
          data-size="sm"
          data-testid="heading-4"
        >
          Don't have an invite yet?
        </h4>
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Ask the Pubky team for your invite code to access the Pubky homeserver.
        </p>
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          A homeserver is a storage provider that hosts your pubky-app-related data. Prefer to use another provider or host yourself? Support for custom homeservers is coming soon.
        </p>
      </div>
      <div
        class="flex-row gap-4"
        data-testid="container"
      >
        <a
          class="text-muted-foreground hover:text-brand"
          data-testid="link"
          href="mailto:hello@pubky.com"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-mail h-6 w-6"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
            />
            <rect
              height="16"
              rx="2"
              width="20"
              x="2"
              y="4"
            />
          </svg>
        </a>
        <a
          class="text-muted-foreground hover:text-brand"
          data-testid="link"
          href="https://x.com/pubky"
        >
          <svg
            class="lucide-x-twitter h-6 w-6"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_18345_202114)"
            >
              <path
                d="M14.2833 10.1616L23.2178 -0.000732422H21.1006L13.3427 8.8231L7.14656 -0.000732422H0L9.36984 13.3425L0 23.9993H2.11732L10.3098 14.681L16.8534 23.9993H24L14.2827 10.1616H14.2833ZM11.3833 13.46L10.4339 12.1313L2.88022 1.55888H6.1323L12.2282 10.0912L13.1776 11.4199L21.1016 22.5106H17.8495L11.3833 13.4605V13.46Z"
                fill="currentColor"
                opacity="0.8"
              />
            </g>
            <defs>
              <clippath
                id="clip0_18345_202114"
              >
                <rect
                  fill="white"
                  height="24"
                  width="24"
                />
              </clippath>
            </defs>
          </svg>
        </a>
        <a
          class="text-muted-foreground hover:text-brand"
          data-testid="link"
          href="https://t.me/pubkychat"
        >
          <svg
            class="lucide-telegram h-6 w-6"
            fill="none"
            height="24"
            viewBox="4 4 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.9678 5.27685C19.9548 5.21777 19.9265 5.16317 19.8856 5.11857C19.8448 5.07397 19.7929 5.04095 19.7351 5.02286C19.5249 4.98123 19.3074 4.99665 19.1052 5.06753C19.1052 5.06753 5.08674 10.1065 4.28612 10.6645C4.11413 10.7845 4.05614 10.8545 4.02747 10.9365C3.88882 11.3365 4.32012 11.5084 4.32012 11.5084L7.93322 12.6857C7.99432 12.6967 8.05714 12.693 8.11655 12.675C8.93783 12.1557 16.3827 7.45271 16.8153 7.29472C16.882 7.27472 16.9333 7.29472 16.92 7.34471C16.748 7.94801 10.3144 13.663 10.2791 13.6976C10.2619 13.7117 10.2485 13.7299 10.2401 13.7505C10.2317 13.7711 10.2286 13.7935 10.2311 13.8156L9.89376 17.3434C9.89376 17.3434 9.75244 18.4413 10.8504 17.3434C11.629 16.5641 12.3763 15.9188 12.7496 15.6049C13.9922 16.4628 15.3288 17.4114 15.9054 17.908C16.0024 18.0019 16.1173 18.0753 16.2433 18.1237C16.3693 18.1722 16.5038 18.1947 16.6387 18.19C16.8048 18.1698 16.9608 18.0994 17.086 17.9883C17.2111 17.8772 17.2995 17.7306 17.3393 17.5681C17.3393 17.5681 19.8925 7.28672 19.9778 5.90947C19.9865 5.77615 19.9978 5.68815 19.9991 5.59549C20.0033 5.48834 19.9927 5.38113 19.9678 5.27685Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`PopoverInviteHomeserver - Snapshots > matches snapshot for default PopoverInviteHomeserver 1`] = `
<div
  data-testid="popover"
>
  <div
    data-as-child="true"
    data-testid="popover-trigger"
  >
    <button
      class="hover:bg-brand/10"
      data-size="icon"
      data-testid="button-ghost"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-circle-help h-4 w-4 text-white"
        data-testid="circle-help-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <path
          d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
        />
        <path
          d="M12 17h.01"
        />
      </svg>
    </button>
  </div>
  <div
    class="w-[327px]"
    data-testid="popover-content"
  >
    <div
      class="flex-col gap-6 px-3 py-2"
      data-testid="container"
    >
      <div
        class="flex-col gap-2"
        data-testid="container"
      >
        <h4
          class="text-popover-foreground"
          data-size="sm"
          data-testid="heading-4"
        >
          Don't have an invite yet?
        </h4>
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Ask the Pubky team for your invite code to access the Pubky homeserver.
        </p>
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          A homeserver is a storage provider that hosts your pubky-app-related data. Prefer to use another provider or host yourself? Support for custom homeservers is coming soon.
        </p>
      </div>
      <div
        class="flex-row gap-4"
        data-testid="container"
      >
        <a
          class="text-muted-foreground hover:text-brand"
          data-testid="link"
          href="mailto:hello@pubky.com"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-mail h-6 w-6"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
            />
            <rect
              height="16"
              rx="2"
              width="20"
              x="2"
              y="4"
            />
          </svg>
        </a>
        <a
          class="text-muted-foreground hover:text-brand"
          data-testid="link"
          href="https://x.com/pubky"
        >
          <svg
            class="lucide-x-twitter h-6 w-6"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_18345_202114)"
            >
              <path
                d="M14.2833 10.1616L23.2178 -0.000732422H21.1006L13.3427 8.8231L7.14656 -0.000732422H0L9.36984 13.3425L0 23.9993H2.11732L10.3098 14.681L16.8534 23.9993H24L14.2827 10.1616H14.2833ZM11.3833 13.46L10.4339 12.1313L2.88022 1.55888H6.1323L12.2282 10.0912L13.1776 11.4199L21.1016 22.5106H17.8495L11.3833 13.4605V13.46Z"
                fill="currentColor"
                opacity="0.8"
              />
            </g>
            <defs>
              <clippath
                id="clip0_18345_202114"
              >
                <rect
                  fill="white"
                  height="24"
                  width="24"
                />
              </clippath>
            </defs>
          </svg>
        </a>
        <a
          class="text-muted-foreground hover:text-brand"
          data-testid="link"
          href="https://t.me/pubkychat"
        >
          <svg
            class="lucide-telegram h-6 w-6"
            fill="none"
            height="24"
            viewBox="4 4 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.9678 5.27685C19.9548 5.21777 19.9265 5.16317 19.8856 5.11857C19.8448 5.07397 19.7929 5.04095 19.7351 5.02286C19.5249 4.98123 19.3074 4.99665 19.1052 5.06753C19.1052 5.06753 5.08674 10.1065 4.28612 10.6645C4.11413 10.7845 4.05614 10.8545 4.02747 10.9365C3.88882 11.3365 4.32012 11.5084 4.32012 11.5084L7.93322 12.6857C7.99432 12.6967 8.05714 12.693 8.11655 12.675C8.93783 12.1557 16.3827 7.45271 16.8153 7.29472C16.882 7.27472 16.9333 7.29472 16.92 7.34471C16.748 7.94801 10.3144 13.663 10.2791 13.6976C10.2619 13.7117 10.2485 13.7299 10.2401 13.7505C10.2317 13.7711 10.2286 13.7935 10.2311 13.8156L9.89376 17.3434C9.89376 17.3434 9.75244 18.4413 10.8504 17.3434C11.629 16.5641 12.3763 15.9188 12.7496 15.6049C13.9922 16.4628 15.3288 17.4114 15.9054 17.908C16.0024 18.0019 16.1173 18.0753 16.2433 18.1237C16.3693 18.1722 16.5038 18.1947 16.6387 18.19C16.8048 18.1698 16.9608 18.0994 17.086 17.9883C17.2111 17.8772 17.2995 17.7306 17.3393 17.5681C17.3393 17.5681 19.8925 7.28672 19.9778 5.90947C19.9865 5.77615 19.9978 5.68815 19.9991 5.59549C20.0033 5.48834 19.9927 5.38113 19.9678 5.27685Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
